.github/workflows/ci-build.yml

```diff
@@ -1,14 +1,52 @@
-name: Test deployment
+name: Format and Test
+
+# Prevent formatting commit collisions when pushing code too fast.
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
 
 on:
   pull_request:
     branches:
       - main
 
 jobs:
-  test-deploy:
+  format:
+    name: Format code
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    outputs:
+      head: ${{ steps.head.outputs.head }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Check out the branch that the PR refers to, so the git HEAD isn't detached.
+          ref: ${{ github.event.pull_request.head.ref }}
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: yarn
+      - run: npx prettier --write .
+      - name: Push changes
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            # github-actions name and email derived from https://github.com/orgs/community/discussions/26560
+            git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git config user.name "github-actions[bot]"
+            git commit -am "Automatic formatting"
+            git push
+          fi
+      - name: Get head SHA
+        id: head
+        run: echo "head=$(git rev-parse HEAD)" >>"$GITHUB_OUTPUT"
+
+  test:
     name: Test deployment
     runs-on: ubuntu-latest
+    permissions:
+      statuses: write
+    needs: [format]
     steps:
       - uses: actions/checkout@v3
```

Comment:
We should probably check out `${{ needs.format.outputs.head }}` instead here, just in case the formatting changes are relevant.